File: api/src/main/java/com/aistore/module/zone/entity/Zone.java
package com.aistore.module.zone.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/** 库区实体（表 zone，隶属仓库） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("zone")
public class Zone {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private Long warehouseId;
    private String code;
    private String name;
    private String type;
    private String remark;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}

File: web/src/types/zone.ts
/* ========================================
   库区模块类型（对齐 OpenAPI 契约）
   ======================================== */

/** 库区详情（对齐 Zone schema） */
export interface Zone {
  id: number;
  warehouseId: number;
  warehouseName?: string | null;
  code: string;
  name: string;
  type?: string | null;
  remark?: string | null;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** 库区列表项（对齐 ZoneSummary schema） */
export interface ZoneSummary {
  id: number;
  warehouseId: number;
  warehouseName?: string | null;
  code: string;
  name: string;
  type?: string | null;
  status: 0 | 1;
  createdAt: string;
}

/** 库区列表分页响应 */
export interface ZoneListResponse {
  items: ZoneSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 创建库区请求 */
export interface CreateZoneRequest {
  warehouseId: number;
  code: string;
  name: string;
  type?: string;
  remark?: string;
  status?: 0 | 1;
}

/** 更新库区请求（warehouseId/code 不可改，状态走独立接口） */
export interface UpdateZoneRequest {
  name?: string;
  type?: string;
  remark?: string;
}

/** 库区查询参数 */
export interface ZoneQueryParams {
  keyword?: string;
  warehouseId?: number;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}
